src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -221,17 +231,45 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PropertyGroup>
 
     <ItemGroup>
-      <!-- Treat any assemblies that already have customized TrimMode as trimmable. -->
+      <!-- Set IsTrimmable for any assemblies that already have customized TrimMode. -->
       <ManagedAssemblyToLink Condition=" '%(ManagedAssemblyToLink.TrimMode)' != '' ">
         <IsTrimmable>true</IsTrimmable>
       </ManagedAssemblyToLink>
-      <!-- Root and copy non-trimmable assemblies. -->
+
+      <!-- SetIsTrimmable for any assemblies listed in TrimmableAssembly. -->
+      <_ManagedAssemblyToLinkName Include="@(ManagedAssemblyToLink->'%(Filename)')">
+        <OriginalItemSpec>%(Identity)</OriginalItemSpec>
+      </_ManagedAssemblyToLinkName>
+      <_NonTrimmableManagedAssemblyToLinkName Include="@(_ManagedAssemblyToLinkName)" Exclude="@(TrimmableAssembly)" />
+      <_TrimmableManagedAssemblyToLinkName Include="@(_ManagedAssemblyToLinkName)" Exclude="@(_NonTrimmableManagedAssemblyToLinkName)" />
+      <_TrimmableManagedAssemblyToLink Include="@(_TrimmableManagedAssemblyToLinkName->'%(OriginalItemSpec)')" />
```

Comment:
I think you might be able to use `JoinItems` task here to reduce some of these shenanigans.  e.g. https://github.com/dotnet/sdk/blob/83d9d6971af6cdf68e409f7f9baa5501093f5018/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets#L434-L446

https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/JoinItems.cs

Reply:
Thanks, this is much nicer!